src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
```

Comment:
PullProofAI

The `calculateAverage` method can be optimized by using the `reduce` function, which is more concise and functional in style.
```diff
  public calculateAverage(numbers: number[]): number {
    if (!numbers.length) return 0;
-4:     let sum = 0;
-5:     for (const num of numbers) {
-6:       sum += num;
-7:     }
-8:     return sum / numbers.length;
+4:     const sum = numbers.reduce((acc, num) => acc + num, 0);
+5:     return sum / numbers.length;
```



<!-- This is an auto-generated comment by PullProofAI -->